From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 24 Oct 2016 19:26:36 +0200
Subject: Further fix widget sizing

Follow-up by ofourdan in order to fix resizing on maximization. This
makes the drawing area ignore configure events that weren't explicitly
sent (such as from GtkForm).
---
 src/gui_gtk_x11.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui_gtk_x11.c b/src/gui_gtk_x11.c
index 64edfd4..eda6dc2 100644
--- a/src/gui_gtk_x11.c
+++ b/src/gui_gtk_x11.c
@@ -3099,6 +3099,9 @@ drawarea_configure_event_cb(GtkWidget	      *widget,
     if (event->width == cur_width && event->height == cur_height)
 	return TRUE;
 
+    if (!gtk_check_version(3, 22, 2) && event->send_event == FALSE)
+	return TRUE;
+
     cur_width = event->width;
     cur_height = event->height;
 
